std/experimental/ndslice/slice.d

```diff
@@ -567,6 +567,19 @@ slice(T,
 }
 
 /// ditto
+auto slice(T,
+    Flag!`replaceArrayWithPointer` replaceArrayWithPointer = Yes.replaceArrayWithPointer,
+    size_t N)(auto ref in size_t[N] lengths, auto ref T init)
+{
+    import std.array : uninitializedArray;
+    immutable len = lengthsProduct(lengths);
+    auto arr = uninitializedArray!(T[])(len);
```

Comment:
`uninitializedArray` should be used  only for types without elaborate constructor